test/integration/integration_admin_test.cc

```diff
@@ -510,39 +506,50 @@ TEST_P(IntegrationAdminTest, AdminDrainInboundOnlyIdempotent) {
       lookupPort("admin"), "POST", "/drain_listeners?inboundonly&graceful", "",
       downstreamProtocol(), version_);
   EXPECT_TRUE(response->complete());
-  EXPECT_EQ("304", response->headers().getStatusValue());
+  EXPECT_EQ("200", response->headers().getStatusValue());
   EXPECT_EQ("text/plain; charset=UTF-8", contentType(response));
-  EXPECT_EQ("already draining\n", response->body());
+  EXPECT_EQ("OK\n", response->body());
 
+  // This should actually cause a new drain since this one isn't inbound_only
   BufferingStreamDecoderPtr response3 = IntegrationUtil::makeSingleRequest(
-      lookupPort("admin"), "POST", "/drain_listeners?graceful", "",
-      downstreamProtocol(), version_);
+      lookupPort("admin"), "POST", "/drain_listeners?graceful", "", downstreamProtocol(), version_);
   EXPECT_TRUE(response->complete());
   EXPECT_EQ("200", response->headers().getStatusValue());
   EXPECT_EQ("text/plain; charset=UTF-8", contentType(response));
   EXPECT_EQ("OK\n", response->body());
 
-  response = IntegrationUtil::makeSingleRequest(lookupPort("inbound_0"), "GET", "/", "",
-                                                downstreamProtocol(), version_);
-  EXPECT_TRUE(response->complete());
-  EXPECT_EQ("close", response->headers().getConnectionValue());
+  codec_client_ = makeHttpConnection(lookupPort("inbound_0"));
+  EXPECT_FALSE(codec_client_->disconnected());
+
+  IntegrationStreamDecoderPtr response4;
+  response4 = codec_client_->makeHeaderOnlyRequest(default_request_headers_);
```

Comment:
nit: should just initialize response4 once vs multiple times.

e.g. here we initialize it to nullptr then do assignment operator.

Doing:

```
IntegrationStreamDecoderPtr response4 =  codec_client_->makeHeaderOnlyRequest(default_request_headers_);
```

inits it once